validator/rpc/handlers_keymanager.go

```diff
@@ -0,0 +1,92 @@
+package rpc
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/gorilla/mux"
+	"github.com/pkg/errors"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/rpc/eth/shared"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	http2 "github.com/prysmaticlabs/prysm/v4/network/http"
+	"github.com/prysmaticlabs/prysm/v4/validator/client"
+	"go.opencensus.io/trace"
+	"google.golang.org/protobuf/types/known/emptypb"
+)
+
+// SetVoluntaryExit creates a signed voluntary exit message and returns a VoluntaryExit object.
+func (s *Server) SetVoluntaryExit(w http.ResponseWriter, r *http.Request) {
+	ctx, span := trace.StartSpan(r.Context(), "validator.keymanagerAPI.SetVoluntaryExit")
+	defer span.End()
+
+	rawPubkey := mux.Vars(r)["pubkey"]
+	if rawPubkey == "" {
+		http2.HandleError(w, "pubkey is required in URL params", http.StatusBadRequest)
+		return
+	}
+
+	pubkey, valid := shared.ValidateHex(w, "pubkey", rawPubkey, fieldparams.BLSPubkeyLength)
+	if !valid {
+		return
+	}
+
+	var epoch primitives.Epoch
+	ok, _, e := shared.UintFromQuery(w, r, "epoch")
+	if !ok {
+		http2.HandleError(w, "syncing_status is not a valid HTTP status code", http.StatusBadRequest)
```

Comment:
This error is incorrect
```suggestion
		http2.HandleError(w, "invalid epoch", http.StatusBadRequest)
```